txsort.go

```diff
@@ -0,0 +1,100 @@
+// TxSort
+// Provides functions for sorting tx inputs and outputs according to BIP LI01
+
+package btcutil
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+// TxSort sorts the inputs and outputs of a tx based on BIP LI01
+// It does not modify the transaction given, but returns a new copy
+// which has been sorted and may have a different txid.
+func TxSort(tx *wire.MsgTx) *wire.MsgTx {
+	txCopy := tx.Copy()
+	sortInputs(txCopy.TxIn)
+	sortOutputs(txCopy.TxOut)
+	return txCopy
+}
+
+// TxIsSorted checks whether tx has inputs and outputs sorted according
+// to BIP LI01.
+func TxIsSorted(tx *wire.MsgTx) bool {
+
+	if !checkSortInputs(tx.TxIn) {
```

Comment:
Same here in regards to just doing `if !sort.IsSorted(sortableInputSlice(tx.TxIn))`.  I don't think there is a readability issue there and it will avoid the function overhead (which likely won't be inlined, unfortunately).